lib/ruby_lsp/requests/base_request.rb

```diff
@@ -31,94 +32,10 @@ def run; end
       # Syntax Tree implements `visit_all` using `map` instead of `each` for users who want to use the pattern
       # `result = visitor.visit(tree)`. However, we don't use that pattern and should avoid producing a new array for
       # every single node visited
-      sig { params(nodes: T::Array[SyntaxTree::Node]).void }
+      sig { params(nodes: T::Array[T.nilable(SyntaxTree::Node)]).void }
```

Comment:
Why do we need this to be nilable?

Authorizing nilables on `visit(node)` makes sense to make it easier to call `visit(node.something)` where `something` may be nil and we don't want to repeat the guard `if node.something`. But in `visit_all` shouldn't we always pass around collections of non-nilable nodes, even if we need to compact them?

If this change coming from the upstream signature maybe?